### PR TITLE
Add a checker if indices might overflow UINT32_MAX

### PR DESCRIPTION
**Pull request by lum1n0us**
[69920](https://oss-fuzz.com/testcase-detail/5082950006276096)

may just pick one between this one and https://github.com/bytecodealliance/wasm-micro-runtime/pull/3577

### CODE REVIEW

File: core/iwasm/interpreter/wasm_loader.c [hunk 1]
```diff
@@ -3627,6 +3627,10 @@ load_function_section(const uint8 *buf, const uint8 *buf_end,
         return false;
     }
 
+    if (is_indices_overflow(module->import_function_count, func_count,
+                            error_buf, error_buf_size))
+        return false;
```
Comment (wenyongh) [inline core/iwasm/interpreter/wasm_loader.c:3632]: Had better add `bh_assert` in wasm_mini_loader.c?

File: core/iwasm/interpreter/wasm_loader.c [hunk 2]
```diff
@@ -4022,6 +4026,9 @@ load_global_section(const uint8 *buf, const uint8 *buf_end, WASMModule *module,
 #endif
 
     read_leb_uint32(p, p_end, global_count);
+    if (is_indices_overflow(module->import_global_count, global_count,
+                            error_buf, error_buf_size))
+        return false;
```
Comment (wenyongh) [inline core/iwasm/interpreter/wasm_loader.c:4031]: ditto

**Review by wenyongh — APPROVED** [review]

File: core/iwasm/common/wasm_loader_common.c [hunk 3]
```diff
@@ -96,3 +96,19 @@ is_valid_func_type(const WASMFuncType *func_type)
...
+/*
+ * Indices are represented as a u32.
+ */
+bool
+is_indices_overflow(uint32 import, uint32 other, char *error_buf,
+                    uint32 error_buf_size)
+{
+    if (import > UINT32_MAX - other) {
+        snprintf(error_buf, error_buf_size,
+                 "too many items in the index space(%u+%u).", import, other);
```
Comment (wenyongh) [inline core/iwasm/common/wasm_loader_common.c:109]: The nuttx CI reported error:
```bash
wamr/core/iwasm/common/wasm_loader_common.c: In function 'is_indices_overflow':
wamr/core/iwasm/common/wasm_loader_common.c:109:54: error: format '%u' expects argument of type 'unsigned int', but argument 4 has type 'uint32' {aka 'long unsigned int'} [-Werror=format=]
  109 |                  "too many items in the index space(%u+%u).", import, other);
      |                                                     ~^        ~~~~~~
      |                                                      |        |
      |                                                      |        uint32 {aka long unsigned int}
      |                                                      unsigned int
      |                                                     %lu
```
https://github.com/bytecodealliance/wasm-micro-runtime/actions/runs/9736851200/job/26869359352#step:7:2392

Here had better change `"%u"` to `"%" PRIu32 `:
https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/iwasm/interpreter/wasm_runtime.c#L3473

**Review by wenyongh — COMMENTED** [review]
LGTM

**Comment by TianlongLiang** [comment]
LGTM

**Comment by yamt** [comment]
can you do the same for other resources as well?
memories, tables, tags.

**Comment by lum1n0us** [comment]
Sure. But since wamr haven't support multi-tables and multi-memories yet, I'll patch tag only.

**Pull request merged**